### DISCUSSION TITLE
Patches

### DISCUSSION THREAD

**Discussion by parvesh88** [bromite/bromite#2552] [Builds]
<pre>
Affects: chromium/script/copy_config.sh

3,26c3
< # Copyright (c) 2012 The Chromium Authors. All rights reserved.
< # Use of this source code is governed by a BSD-style license that can be
< # found in the LICENSE file.
< 
< # Use this to copy all config files into the tree.
< for os in android linux linux-noasm mac win; do
<   for target in Chromium Chrome ChromeOS; do
<     # Copy config files for various architectures:
<     #   - ia32/x64 have config.asm, config.h
<     #   - arm/arm-neon have config.h
<     for arch in arm arm-neon arm64 ia32 x64 mipsel mips64el; do
<       # Don't waste time on non-existent configs, if no config.h then skip.
<       [ ! -e "build.$arch.$os/$target/config.h" ] && continue
<       for f in config.h config_components.h config.asm libavutil/avconfig.h libavutil/ffversion.h libavcodec/bsf_list.c libavcodec/codec_list.c libavcodec/parser_list.c libavformat/demuxer_list.c libavformat/muxer_list.c libavformat/protocol_list.c; do
<         FROM="build.$arch.$os/$target/$f"
<         TO="chromium/config/$target/$os/$arch/$f"
<         if [ "$(dirname $f)" != "" ]; then mkdir -p $(dirname $TO); fi
<         [ -e $FROM ] && cp -v $FROM $TO
<       done
<     done
<   done
< done
< 
< echo "Copied all existing newer configs successfully."
---
> echo "Skipped Copying all existing newer configs."

Affects: chromium/script/build_ffmpeg.py

726a727,729
>         '--enable-decoders',
>         '--enable-parsers',
>         '--enable-demuxers',

</pre>

**Comment by uazo** [comment]
what is it?

**Reply by parvesh88** [reply]
difference of file A and file B

**Reply by uazo** [reply]
OK, but the purpose of that modification? what does it do? why did you write it?

**Reply by parvesh88** [reply]
enable all ffmpeg decoders and skip copying google specified ffmpeg configs

**Reply by uazo** [reply]
and why do you disagree with the configurations entered by google?

**Reply by parvesh88** [reply]
You have to add all libavcodec, libavutil and libavformat sources in ffmpeg_generated.gni file

**Reply by parvesh88** [reply]
> and why do you disagree with the configurations entered by google?

because it have limited codecs support

**Reply by uazo** [reply]
[third_party/ffmpeg/chromium/scripts/copy_config.sh](https://source.chromium.org/chromium/chromium/src/+/main:third_party/ffmpeg/chromium/scripts/copy_config.sh;bpv=1) appears to be a service script launched manually with each roll of ffmpeg updates, and from the change history it appears to be used to configure the ffmpeg build by copying from a folder that not exists (probably a development location), so why the changes?

so [ffmpeg/chromium/config](https://source.chromium.org/chromium/chromium/src/+/main:third_party/ffmpeg/chromium/config/;bpv=1) seem to be the current configurations used by the build, with or without the changes to copy_config.sh

perhaps the other changes enable the use of all codes https://source.chromium.org/chromium/chromium/src/+/main:third_party/ffmpeg/chromium/scripts/build_ffmpeg.py;l=726

but I still don't understand, why are you writing them here and in this way?

**Reply by uazo** [reply]
rather [they](https://source.chromium.org/chromium/chromium/src/+/main:third_party/ffmpeg/chromium/config/;bpv=1) seem to be the result of 'make configure' which is probably inserted so as not to launch it during the chromium build, to make it faster I think.

**Reply by parvesh88** [reply]
Ok
but you have to add more other codecs references in codecs_list 
see https://source.chromium.org/chromium/chromium/src/+/main:third_party/ffmpeg/chromium/config/Chrome/android/arm-neon/libavcodec/codec_list.c

**Reply by uazo** [reply]
why? that is, but, in the case of adding codecs, what is the benefit to users of a browser capable of using them? in which sites are there videos that today's browser cannot read?

**Reply by parvesh88** [reply]
> why? that is, but, in the case of adding codecs, what is the benefit to users of a browser capable of using them? in which sites are there videos that today's browser cannot read?

User can watch and listen all type of videos i.e. MPEG2TS and audios i.e. AC3 and EAC3

**Reply by uazo** [reply]
okay, even if that were the case (but enabling it in the library doesn't necessarily mean enabling them on the browser, there could be other checks elsewhere before you get to the library), why propose it (is that what you're doing?) in a browser more related to privacy issues? possibly the issue could be the reverse, i.e., removing codecs rather than adding them. that's what I don't understand.

also think of it the other way around: what if enabling them all brings more disadvantages (in terms of privacy) than advantages? with what tools could we prevent this?

**Reply by parvesh88** [reply]
> okay, even if that were the case (but enabling it in the library doesn't necessarily mean enabling them on the browser, there could be other checks elsewhere before you get to the library), why propose it (is that what you're doing?) in a browser more related to privacy issues? possibly the issue could be the reverse, i.e., removing codecs rather than adding them. that's what I don't understand.
> 
> also think of it the other way around: what if enabling them all brings more disadvantages (in terms of privacy) than advantages? with what tools could we prevent this?

Ok, I agreed with you that there can be other checks to enable them on browser and privacy related issues but we can add most commonly used video and audio codecs in the library

**Reply by uazo** [reply]
> and privacy related issues

However, for bromite, that is exactly the problem.
assuming that you allow users to be able to view any stream, i know, cast by vlc server, since those options being in the build phase, would they be available to all accessible websites, and if among the codecs that you enable is there anything that can be exploited? i would imagine that even ffmpeg might have its cve, which might not be verified by the google team because maybe not applicable with respect to the configuration that they chose in chrome.
it may be too risky and not very beneficial

**Reply by parvesh88** [reply]
> > and privacy related issues
> 
> However, for bromite, that is exactly the problem. assuming that you allow users to be able to view any stream, i know, cast by vlc server, since those options being in the build phase, would they be available to all accessible websites, and if among the codecs that you enable is there anything that can be exploited? i would imagine that even ffmpeg might have its cve, which might not be verified by the google team because maybe not applicable with respect to the configuration that they chose in chrome. it may be too risky and not very beneficial

Ok
But can you not verified ffmpeg cve in chromium test build at your own level?
